sparkplug/timereporters/ddstatsd.py

```diff
@@ -0,0 +1,69 @@
+"""
+Sends timing information to Datadog custom metrics
+Uses a thread to send at regular intervals, without
+interfering with the main consumer thread
+"""
+
+from sparkplug.logutils import LazyLogger
+_log = LazyLogger(__name__)
+
+from sparkplug.timereporters.base import Base, mn_md_mx
+
+try:
+    # pip install datadog
+    from datadog import initialize
+
+    from datadog.threadstats.base import ThreadStats
+
+    is_initialized = False
+
+    class DDStatsd(Base):
+        def __init__(self, aggregation_count, api_key, app_key, tags=None):
+            super().__init__(aggregation_count)
+            global is_initialized
+            if not is_initialized:
+                initialize(api_key=api_key, app_key=app_key)
+                is_initialized = True
+            self.exec = []
+            self.erro = []
+            self.wait = []
+            self.tags = self._parse_tags(tags)
+            self.statsd = ThreadStats()
+            self.statsd.start()
+
+        def _parse_tags(self,tags):
+            ret = None
```

Comment:
What happens when we send statsd metrics with the tags argument as None? I wonder if we should be defaulting the return value to an empty list. It might simplify the code block below.

Reply:
I'm trying to stick to the convention in the datadog documents.  When you're not associating tags, they default to None.